// aws-sdk-js-codemod/dist/transforms/v2-to-v3/apis/replaceS3UploadApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.replaceS3UploadApi = void 0;
const getClientIdentifiers_1 = require("./getClientIdentifiers");
const getS3UploadCallExpression_1 = require("./getS3UploadCallExpression");
// Updates `s3.upload()` API with `new Upload()` API.
const replaceS3UploadApi = (j, source, options) => {
    if (options.v2ClientName !== "S3")
        return;
    const clientIdentifiers = (0, getClientIdentifiers_1.getClientIdentifiers)(j, source, options);
    for (const clientId of clientIdentifiers) {
        source
            .find(j.CallExpression, (0, getS3UploadCallExpression_1.getS3UploadCallExpression)(clientId))
            .replaceWith((callExpression) => {
            const params = callExpression.node.arguments[0];
            const properties = [];
            properties.push(j.objectProperty.from({
                key: j.identifier("client"),
                value: clientId,
                shorthand: true,
            }));
            if (params) {
                properties.push(j.objectProperty.from({
                    key: j.identifier("params"),
                    // @ts-expect-error Type 'SpreadElement | ExpressionKind' is not assignable
                    value: params,
                    shorthand: true,
                }));
            }
            const options = callExpression.node.arguments[1];
            if (options) {
                switch (options.type) {
                    case "ObjectExpression":
                        properties.push(...options.properties);
                        break;
                    case "Identifier":
                        properties.push(j.spreadElement(options));
                        break;
                }
            }
            return j.newExpression(j.identifier("Upload"), [j.objectExpression(properties)]);
        });
        // Replace `.promise()` call with `.done()` if present.
        source
            .find(j.MemberExpression, {
            type: "MemberExpression",
            object: {
                type: "NewExpression",
                callee: { type: "Identifier", name: "Upload" },
            },
            property: { type: "Identifier", name: "promise" },
        })
            .forEach((memberExpression) => {
            memberExpression.value.property.name = "done";
        });
    }
};
exports.replaceS3UploadApi = replaceS3UploadApi;
